### ISSUE TITLE
Illegal Instruction exception due to overlapping registers

### ISSUE THREAD

**Issue by zeeshandildar** [riscv-non-isa/riscv-arch-test#500]
I have encountered illegal instruction exception due to overlapping registers in vector crypto test i.e vsha2ms-e32.vv-01 (line 73). Replacing that with other register resolved the issue
```
inst_8:
TEST_CASE_WVV(4, 32, VINST, v16, 0*4, v17, 4*4, v16, 0*4)
//sig[120*4]
```

**Comment by allenjbaum** [comment]
There should be some tests that check whether an illegal instruction
encoding would cause an exception. could this be one of them?
IF not, then we have to ask the developers why this test failure wasn't
caught in their testing of the code.

**Comment by zeeshandildar** [comment]
I am not sure if this was intended to generate exception

**Comment by allenjbaum** [comment]
We should probably consult with the author to confirm their intent before
accepting this PR, then

**Comment by allenjbaum** [comment]
Oh, and if a trap was not the intent, we have to ask how this happened -
this should have been caught somewhere.
Another possibility is that this is optional behavior, and it was tested on
models that exhibit just one of them.

**Comment by allenjbaum** [comment]
My understanding is that none of these tests are expected to trap, so these
tests will likely need to be regenerated to avoid this case.
There should also be separate tests that check for this condition
(unless those already exist for other vector ops,  and we deem it
unnecessary to test it for all possible ops, which is probably the case).
We do need these tests for all vector op classes at least, e.g. .VV,.VI,
etc.

Generally (for base ops that can trap) these are put into the "privileged"
test suite, with the trap handler enabled.
We should figure out how to catch unexpected traps if the trap handler is
not instantiated (effectively installing a much simpler handler that does
nothing but end the test?).
If the trap handler is installed, we should shorten the default trap
signature area by a word.
That will cause the regular trap handler to end the test because it checks
for trap signature overflow.
Unfortunately, that only works if the trap signature area is defined to be
exactly the right size - and that isn't always possible,
as different implementations may trap or not depending on the architectural
options it supports).

**Comment by UmerShahidengr** [comment]
Merged #501 to resolve this issue. Closing this one, but it can be opened again anytime
